### PR TITLE
Done

### PR DESCRIPTION
**Pull request by stefanT9**
Functioneaza doar ca nu stiu daca e metoda corecta pe partea de register. Am modificat direct in request.

### CODE REVIEW

File: controllers/authController.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 const HttpStatus = require('http-status-codes')
 const { createTkn } = require('../utils')
 const sendEmail = require('../utils/sendMail')
+const bcrypt = require('bcrypt')
+const saltRounds = 10
```
Comment (procateodor) [inline controllers/authController.js]: Let's move all the constants like this under `utils/constants`

File: controllers/authController.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+const salt = bcrypt.genSaltSync(saltRounds)
```
Comment (procateodor) [inline controllers/authController.js]: We don't need to generate a salt, will be autogenerated. see technique 2 from bcrypt.

File: controllers/authController.js [hunk 3]
```diff
@@ -13,7 +16,7 @@ exports.login = async (req, res) => {
       })
     }
```
Comment (procateodor) [inline controllers/authController.js:16]: just use the `compare` function to check that

File: controllers/authController.js [hunk 4]
```diff
@@ -50,6 +53,7 @@ exports.register = async (req, res) => {
...
 
+    req.body.password = bcrypt.hashSync(req.body.password, salt)
```
Comment (procateodor) [inline controllers/authController.js]: here we should use the technique 2 to hash this w/o salt, using only the salt rounds.

**Review by procateodor — CHANGES_REQUESTED** [review]
Also, we have to add encryption to the `change_password` route too.

**Comment by procateodor** [comment]
[Password encryption](https://trello.com/c/7J7aOKYY/17-password-encryption)

File: controllers/authController.js [hunk 5]
```diff
@@ -13,7 +14,8 @@ exports.login = async (req, res) => {
       })
     }
 
-    if (req.body.password !== user.password) {
+    req.body.password = bcrypt.hashSync(req.body.password, constants.saltRounds)
```
Comment (procateodor) [inline controllers/authController.js]: No need of this

**Review by procateodor — CHANGES_REQUESTED** [review]
Looks good, just a minor comment

File: controllers/authController.js (hunk 5 above)
Comment (stefanT9) [reply controllers/authController.js]: Daca scot linia aia nu va lasa in baza de date parola in plaintext?

Creerea userului se face cu  `const user = await req.db.User.create(req.body)`

Comment (procateodor) [reply controllers/authController.js]: @stefanT9 compare compara un Plain text cu un hash. Req.body.password poate fi Plain la compare

**Review by procateodor — DISMISSED** [review]

**Review by procateodor — DISMISSED** [review]

**Review by procateodor — APPROVED** [review]

**Pull request merged**